### PR TITLE
Normalize kernel version in labels and image tags

### PR DESCRIPTION
**Pull request by qbarrand**
Handle uncommon kernel versions with characters such as `+`, by replacing them with `_`.

/cc @ybettan @mresvanis

### CODE REVIEW

File: internal/kernel/kernel.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package kernel
+
+import "strings"
+
+func replaceInvalidChar(r rune) rune {
+	if (r < 'A' || r > 'Z') && (r < 'a' || r > 'z') && (r < '0' || r > '9') && r != '-' && r != '.' && r != '_' {
+		return '_'
+	}
```
Comment (ybettan) [inline internal/kernel/kernel.go:8]: Shouldn't we just replace `+` with `_`?

File: internal/api/moduleloaderdata.go [hunk 2]
```diff
@@ -16,6 +16,11 @@ import (
 type ModuleLoaderData struct {
 	// kernel version
 	KernelVersion string
+
+	// KernelNormalizedVersion is the kernel version with some characters replaced with '_' so that it can be used in
+	// a Kubernetes label or a container image tag.
+	KernelNormalizedVersion string
```
Comment (ybettan) [inline internal/api/moduleloaderdata.go:22]: Do we still need `KernelVersion`?

File: internal/kernel/kernel.go (hunk 1 above)
Comment (qbarrand) [reply internal/kernel/kernel.go:8]: Handling any invalid character makes the code more future-proof than just handling `+`.

File: internal/api/moduleloaderdata.go (hunk 2 above)
Comment (qbarrand) [reply internal/api/moduleloaderdata.go:22]: We do, in `module_nmc_reconciler.go` for instance.

**Comment by yevgeny-shnaidman** [comment]
do we still need that line:
https://github.com/kubernetes-sigs/kernel-module-management/blob/582546abe7a94a82fbb9062f9c6d713da031df98/internal/controllers/module_nmc_reconciler.go#L316

**Comment by qbarrand** [comment]
@yevgeny-shnaidman I think this new solution could make that line redundant, but if we remove it, we change the behaviour.
Some kernels have a trailing `+`, indicating that the source has been patched before compilation. In most cases, you can still build your kmod against the original headers for that kernel.
If we remove that line, we break the Dockerfile of users who assume that they can build their kmod for a modified kernel using the original headers for that version.

**Pull request merged**